### DISCUSSION TITLE
what are the vstorage root keys other than 'published'?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#9115] [Q&A]
I know about `published` for `chainStorage`, but for reference, what are the rest of these?

```
# agd query vstorage keys ''
children:
- activityhash
- beansOwing
- bundles
- egress
- highPrioritySenders
- mailbox
- published
```

- activityhash - something about blockchain consensus. but I can't find the code that defines it, let alone any docs
- beansOwing - found that in [swingset/keeper/keeper.go](https://github.com/Agoric/agoric-sdk/blob/003f0c2232815a8d64a3f9a5b05521a10160ce34/golang/cosmos/x/swingset/keeper/keeper.go). Do we have a text write-up? We recently did one in [inter protocol fees](https://docs.inter.trade/inter-protocol-system-documentation/fees) docs.
- bundles - that's re `agd tx swingset install-bundle`; are there existing docs?
- egress - mailbox stuff; "reserved for future use" I gather
- highPrioritySenders - used for oracles and such, I think...
- mailbox - more future use mailbox stuff

cc @gibson042

**Comment by gibson042** [comment]
Yeah, I think you've pretty much got it, although it's worth noting that everything except `highPrioritySenders` (and maybe that as well) predates `published`, which I introduced to avoid collisions at the top level. Details that I know about:
* `activityhash` is a SwingSet concept; it is modified by [`emitCrankHashes`](https://github.com/Agoric/agoric-sdk/blob/1cd6408e27271160227f8aaa12d9771ea73a1da1/packages/swing-store/src/swingStore.js#L414-L443) after every delivery.
* For `highPrioritySenders`, the rubber meets the road in [`routeAction`](https://github.com/Agoric/agoric-sdk/blob/master/golang/cosmos/x/swingset/keeper/msg_server.go#L30-L41) (which pushes wallent spend actions from such addresses into a distinct comsic-swingset inbound queue that gets [drained before updating timers and processing non-priority actions](https://github.com/Agoric/agoric-sdk/blob/1cd6408e27271160227f8aaa12d9771ea73a1da1/packages/cosmic-swingset/src/launch-chain.js#L638-L669)).

I'm not aware of docs on any of them, nor even which are actively in use.

**Comment by dckc** [comment]
I don't see how `emitCrankHashes` affects vstorage. What am I missing, @gibson042 ?

**Reply by gibson042** [reply]
Ah yeah, that was removed by https://github.com/Agoric/agoric-sdk/pull/7225 . The chain storage entry still exists, it's just a fossil frozen in time.

**Comment by dckc** [comment]
closing in favor of https://github.com/Agoric/documentation/pull/1016
